app/models/vm_or_template.rb

```diff
@@ -1761,13 +1761,25 @@ def self.reconfigurable?(ids)
     vms.all?(&:reconfigurable?)
   end
 
+  PUBLIC_TEMPLATE_CLASSES = %w(ManageIQ::Providers::Openstack::CloudManager::Template).freeze
```

Comment:
I don't understand why the base class is knowledgeable about a public template class from openstack.  Why does this need to be in the base class?

> because it breaks filtering in UI, problem is in model inheritance. 

I don't know what this means.  can you please explain?

Reply:
Hi, @lpichler explained here https://github.com/ManageIQ/manageiq/pull/18128#issuecomment-445298459 why inheritance breaks RBAC filtering for UI, also when using explorer view the RBAC call will be made for the base model, which is vm_or_template. We are trying to apply some custom RBAC rules for openstack provider, so we placed the RBAC call in base model and we want to not change it's behaviour for anything except ```ManageIQ::Providers::Openstack::CloudManager::Template```

@lpichler correct me if I'm wrong :)

Reply:
@alexander-demichev maybe you can add https://github.com/ManageIQ/manageiq/pull/18128#issuecomment-445298459 to the description.

@jrafanie if we will leave this RBAC rule (it is restriction) only in TemplateCloud model, we will get this restriction only for API call

`/api/<request_for_list_of_template_cloud>`

but it will not prevent list of restricted `TemplateCloud`s(restricted by this rule) in this api call:

`/api/<request_for_list_of_vm_or_template>` - which can any client call.


so to have it in base class is about security and basically it implies general RBAC rule (cc @himdel ):
so that it has a basic RBAC rule (learned from @himdel):
`Do not implement RBAC behavior in special models beucase api can call basic models`

Reply:
Indeed, `/api/vms` will ignore any restriction defined in subclasses of `Vm`.

I think the same goes for `Vm.where(...)`

Reply:
Thanks for the explanation. 

For a future PR... would it make sense to have the subclass register this class into the base class?  It seems odd to me for a base class to know about a subclass.

Reply:
Sounds reasonable :+1: (though I don't know that much about RBAC internals)

Some kind of registration mechanism would keep the logic in the right place, while allowing it to work even when asking for the basest class.

Reply:
thanks @jrafanie I am making a note for it.